pkgs/top-level/lua-packages.nix

```diff
@@ -270,22 +299,38 @@ let
 
   luaposix = buildLuaPackage rec {
     name = "posix-${version}";
-    version = "33.4.0";
+    version = "34.0.4";
 
     src = fetchFromGitHub {
       owner = "luaposix";
       repo = "luaposix";
       rev = "release-v${version}";
-      sha256 = "0y531p54lx2yf243bcsyp6sv8fvbqidp20yry0xvb85p8zw9dlrq";
+      sha256 = "0p5583vidsm7s97zihf47c34vscwgbl86axrnj44j328v45kxb2z";
     };
 
-    buildInputs = [ perl ];
+    propagatedBuildInputs = [ std.normalize bit32 ];
+
+    buildPhase = ''
+      ${lua}/bin/lua build-aux/luke \
+        package="luaposix" \
+        version="${version}" \
+        LUA="${lua}/bin/lua" \
+        LUA_INCDIR="-I${lua}/include" \
+        INST_LIBDIR="$out/lib/lua/${lua.luaversion}" \
+        INST_LUADIR="$out/share/lua/${lua.luaversion}"
```

Comment:
I was able to remove the last 4 lines here. Are they really necessary?

Reply:
mhm. Version should be there: https://github.com/luaposix/luaposix/blob/649454019dcc983eb6b398624e5b6c68d72f7808/lib/posix/version.lua.in

Reply:
nevermind it is still there.